PaddingRNN/lstm_paddle/train.py

```diff
@@ -174,40 +174,64 @@ def main():
         )
         max_epoch = 1
 
+    # define train program
     main_program = fluid.Program()
     startup_program = fluid.Program()
     if args.enable_ce:
         startup_program.random_seed = SEED
-
     with fluid.program_guard(main_program, startup_program):
-        # Training process
-        loss, last_hidden, last_cell, feed_order = lm_model.lm_model(
-            hidden_size,
-            vocab_size,
-            batch_size,
-            num_layers=num_layers,
-            num_steps=num_steps,
-            init_scale=init_scale,
-            dropout=dropout, 
-            rnn_model=rnn_model)
-
-        # clone from default main program and use it as the validation program
-        inference_program = fluid.default_main_program().clone(for_test=True)
-
-        #print(inference_program)
-
-        fluid.clip.set_gradient_clip(
-            clip=fluid.clip.GradientClipByGlobalNorm(clip_norm=max_grad_norm))
-
-        learning_rate = fluid.layers.create_global_var(
-            name="learning_rate",
-            shape=[1],
-            value=1.0,
-            dtype='float32',
-            persistable=True)
-
-        optimizer = fluid.optimizer.SGD(learning_rate=learning_rate)
-        optimizer.minimize(loss)
+        with fluid.unique_name.guard():
+            if args.use_py_reader and not args.inference_only:
+                loss, last_hidden, last_cell, feed_order, py_reader = lm_model.lm_model(
+                    hidden_size,
+                    vocab_size,
+                    batch_size,
+                    num_layers=num_layers,
+                    num_steps=num_steps,
+                    init_scale=init_scale,
+                    dropout=dropout, 
+                    rnn_model=rnn_model,
+                    use_py_reader=True)
+            else:
+                loss, last_hidden, last_cell, feed_order = lm_model.lm_model(
+                    hidden_size,
+                    vocab_size,
+                    batch_size,
+                    num_layers=num_layers,
+                    num_steps=num_steps,
+                    init_scale=init_scale,
+                    dropout=dropout, 
+                    rnn_model=rnn_model,
+                    use_py_reader=False)
```

Comment:
The above code should be polished, for example, 
```
vars = lm_model.lm_model(
                    hidden_size,
                    vocab_size,
                    batch_size,
                    num_layers=num_layers,
                    num_steps=num_steps,
                    init_scale=init_scale,
                    dropout=dropout, 
                    rnn_model=rnn_model,
                    use_py_reader=args.use_py_reader)
if args.use_py_reader:
      py_reader = vars[-1]
      vars = []
loss, last_hidden, last_cell, feed_order = vars
```

Reply:
Done.